packages/amplify_core/pubspec.yaml

```diff
@@ -13,17 +13,17 @@ dependencies:
   flutter:
     sdk: flutter
   intl: ^0.17.0
-  json_annotation: ^4.3.0
+  json_annotation: ">=4.4.0 <4.5.0"
```

Comment:
I had seen the Dart devs do this somewhere and I figured it meant they might add a change which could screw up things. I have to do this with built_value_generator because the changes in the generated output can throw off some CI checks, for example. Looking at where I'd seen it for this, though, it seems like they're back to caret syntax, so I'll update it here and throughout.